src/gui/properties/trackerlist.cpp

```diff
@@ -245,20 +252,30 @@ void TrackerList::loadStickyItems(BitTorrent::TorrentHandle *const torrent) {
     lsd_item->setText(COL_MSG, privateMsg);
   }
 
-  // XXX: libtorrent should provide this info...
-  // Count peers from DHT, LSD, PeX
-  uint nb_dht = 0, nb_lsd = 0, nb_pex = 0;
-  foreach (const BitTorrent::PeerInfo &peer, torrent->peers()) {
-    if (peer.fromDHT())
-      ++nb_dht;
-    if (peer.fromLSD())
-      ++nb_lsd;
-    if (peer.fromPeX())
-      ++nb_pex;
-  }
-  dht_item->setText(COL_RECEIVED, QString::number(nb_dht));
-  pex_item->setText(COL_RECEIVED, QString::number(nb_pex));
-  lsd_item->setText(COL_RECEIVED, QString::number(nb_lsd));
+    // XXX: libtorrent should provide this info...
+    // Count peers from DHT, PeX, LSD
+    uint seedsDHT = 0, seedsPeX = 0, seedsLSD = 0, peersDHT = 0, peersPeX = 0, peersLSD = 0;
+    foreach (const BitTorrent::PeerInfo &peer, torrent->peers()) {
+        if (peer.fromDHT() && peer.isSeed())
+            ++seedsDHT;
+        else if (peer.fromDHT() && !peer.isConnecting())
+            ++peersDHT;
+        if (peer.fromPeX() && peer.isSeed())
+            ++seedsPeX;
+        else if (peer.fromPeX() && !peer.isConnecting())
+            ++peersPeX;
+        if (peer.fromLSD() && peer.isSeed())
+            ++seedsLSD;
+        else if (peer.fromLSD() && !peer.isConnecting())
```

Comment:
I added the `!peer.isConnecting()` because otherwise these numbers are bloated in the beginning and they turn out to be seeds or not connecting to many of them at all.
Let me know if you want me to remove it.